### ISSUE TITLE
Length is returning the paginated length

### ISSUE THREAD

**Issue by ChristopherKiss** [Colonise/DataSource#3]
Length should only consider filtered results

Pagination and sorting should not affect the total length of the dataSource

**Comment by pathurs** [comment]
I think it's best to expose both the filtered length and the paginated length.

**Comment by ChristopherKiss** [comment]
As discussed:

totalLength - Unfiltered and unpaged count.
filteredLength - Filtered and unpaged count.
pageLength - Filtered and paged count.

Or whatever naming you decide upon.

**Comment by pathurs** [comment]
Fixed in #4 

New API:

```
const tableDataSource = new TableDataSource([1, 2, 3]);

const totalLength = tableDataSource.length;
const filteredLength = tableDataSource.filtering.length;
const sortedLength = tableDataSource.sorting.length; // Same as filteredLength
const pageLength = tableDataSource.paging.length;
```
